keychain.h

```diff
@@ -27,36 +27,92 @@
 
 #include <string>
 
+/*! \brief A thin wrapper to provide cross-platform access to the operating
+ *         system's credentials storage.
+ *
+ * keychain provides the functions getPassword, setPassword, and deletePassword.
+ *
+ * All of these functions require three input parameters to identify the
+ * credential that should be retrieved or manipulated: package, service, and
+ * user. These identifiers will be mangled differently on each OS to correspond
+ * to the OS API.
+ * While none of the supported OSes has specific requirements to the format
+ * identifiers, the reverse domain name format is recommended for the package
+ * parameter in order to correspond with conventions.
```

Comment:
I think you want to:

1) mark the `package` parameter with back-ticks to highlight that "package" is an identifier used below
2) repeat that (in short) in each of the method documentations